src/Microsoft.AspNet.Server.Kestrel/Http/Frame.cs

```diff
@@ -426,40 +425,38 @@ public void ProduceEnd(Exception ex)
             var hasConnection = false;
             var hasTransferEncoding = false;
             var hasContentLength = false;
-            if (headers != null)
+
+            foreach (var header in _responseHeaders)
```

Comment:
I tried changing that and it went very strange. Had an alternative approach that did work but was ugly; so will wait till this works a bit better.